app/src/main/java/me/guillaumin/android/osmtracker/gpx/TrackFileProvider.java

```diff
@@ -0,0 +1,121 @@
+package me.guillaumin.android.osmtracker.gpx;
+
+import android.content.ContentProvider;
+import android.content.ContentValues;
+import android.database.AbstractCursor;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.ParcelFileDescriptor;
+import android.util.Log;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+
+/**
+ */
+
+public final class TrackFileProvider extends ContentProvider {
+    private static final String TAG = ExportToTempFileTask.class.getSimpleName();
+
+    @Override
+    public boolean onCreate() {
+        return false;
+    }
+
+    @Override
+    public Cursor query(Uri uri, String[] projection, String selection, String[] selectionArgs, String sortOrder) {
+        return new AbstractCursor() {
+            @Override
+            public int getCount() {
+                return 0;
+            }
+
+            @Override
+            public String[] getColumnNames() {
+                return new String[0];
+            }
+
+            @Override
+            public String getString(int column) {
+                return null;
+            }
+
+            @Override
+            public short getShort(int column) {
+                return 0;
+            }
+
+            @Override
+            public int getInt(int column) {
+                return 0;
+            }
+
+            @Override
+            public long getLong(int column) {
+                return 0;
+            }
+
+            @Override
+            public float getFloat(int column) {
+                return 0;
+            }
+
+            @Override
+            public double getDouble(int column) {
+                return 0;
+            }
+
+            @Override
+            public boolean isNull(int column) {
+                return false;
+            }
+        };
+    }
+
+    @Override
+    public String getType(Uri uri) {
```

Comment:
I'm sorry for overlooking this one, surely we should provide right mime-type information.